R/intervention.R

```diff
@@ -0,0 +1,41 @@
+#' Set up intervention for simulation
+#'
+#' @description
+#' `intvn_scale_r0()` is a helper for the \code{simulation_*} functions. It
+#' modifies the relevant arguments of the offspring distribution in order to
+#' mimic the impact of an intervention. In particular, it scales the mean of
+#' the offspring distribution. Currently, it can only handle the poisson and
+#' negative binomial distributions and errors when other offspring
+#' distributions are specified alongside `intvn_scale_r0`.
+#'
+#' @inheritParams simulate_tree
+#' @param r0_reduction The intervention impact. A scalar between 0 and 1.
+#' Scales the mean of `offspring_dist`. `r0_reduction` = 0 implies
+#' no intervention impact and `r0_reduction` = 1 implies full impact.
+#' @param pars_list Parameter(s) for poisson or negative binomial offspring
+#' distribution.
+#' @return List of the offspring distribution parameter(s) with the mean
+#' scaled by \code{1 - intvn_scale_r0}.
+#' @details
+#' `intvn_scale_r0()` scales the mean of the offspring distribution
+#' by \eqn{1 - r0\_reduction} so that the new mean is given as:
+#' \deqn{(1 - r0\_reduction) \times R_0,} where \eqn{R_0} is the
+#' mean of the poisson and negative binomial distribution.
```

Comment:
Perhaps mention that this corresponds to the population-level reduction of R in Lloyd-Smith et al., as opposed to an individual-level reduction (X% of people have R0 set to 0). Eventually we might also want to implement the individual-level reduction, even though that is probably more tricky.

Reply:
> Perhaps mention that this corresponds to the population-level reduction of R in Lloyd-Smith et al., as opposed to an individual-level reduction (X% of people have R0 set to 0).

Oh, good point. I'll add that. 

> Eventually we might also want to implement the individual-level reduction, even though that is probably more tricky.

I am brainstorming in #3 how to go about that to capture generic use cases.